hicommonwealth/commonwealth
# Optimize balance fetching for gated actions

## Link to Issue
Closes: #6012

## Description of Changes
- Optimizes gated actions to fetch balances *only* for stale memberships– this will ultimately make gated actions snappier because it prevents unnecessary fetching

## "How We Fixed It"
- Check which memberships are fresh/stale before performing batch fetch

## Test Plan
- Set the following envs:
    - `MEMBERSHIP_REFRESH_TTL_SECONDS=15`
    - `TBC_BALANCE_TTL_SECONDS=60`
- Truncate the `Memberships` table
- Do the following in a window of less than 15 seconds
    - As a non-admin user– on a thread that’s gated and you are allowed access– add a comment
    - Confirm that 1 or more new Membership records were created in the DB– take note of the `last_checked` values
    - Immediately add a new comment to the thread, confirm that the `last_checked` values of the memberships did NOT change
- Wait 15ish seconds
- Add a new comment to the thread– confirm that the `last_checked` values DID change


## Deployment Plan
N/A

## Other Considerations
N/A
